python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1525,28 +1525,32 @@ def test_torch_tensor_nccl_all_reduce_diff_comms(ray_start_regular):
     from ray.dag.collective_node import CollectiveOutputNode
     from ray.experimental.channel import ChannelContext
 
-    comm_ids = set()
+    nccl_group_ids = set()
     for _, exec_tasks in compiled_dag.actor_to_executable_tasks.items():
         for exec_task in exec_tasks:
             dag_node = compiled_dag.idx_to_task[exec_task.task_idx].dag_node
             if isinstance(dag_node, CollectiveOutputNode):
                 assert exec_task.collective_group is not None
                 nccl_group_id = exec_task.collective_group._type_hint._nccl_group_id
                 assert nccl_group_id is not None
-                comm_ids.add(nccl_group_id)
+                nccl_group_ids.add(nccl_group_id)
 
     # Exactly 2 NCCL groups should be created.
-    assert len(comm_ids) == 2
+    assert len(nccl_group_ids) == 2
     ctx = ChannelContext.get_current()
-    comms = [ctx.nccl_groups[comm_id] for comm_id in comm_ids]
-    comm_actors = [comm.get_actor_handles() for comm in comms]
-    # One of the NCCL group should contain workers[0] and
-    # the other should contain workers[1]
-    assert {frozenset(comm_actor) for comm_actor in comm_actors} == {
-        frozenset([worker]) for worker in workers
-    }
-
-    # Sanity check: No p2p communicator constructed.
+    nccl_groups = [ctx.nccl_groups[nccl_group_id] for nccl_group_id in nccl_group_ids]
+    nccl_actor_lists = [nccl_group.get_actor_handles() for nccl_group in nccl_groups]
+    # One of the NCCL group should contain workers[0] and the other should
+    # contain workers[1].
+    actor_groups = set()
+    for nccl_actor_list in nccl_actor_lists:
+        actor_groups.add(frozenset(nccl_actor_list))
+    expected_actor_groups = set()
+    for worker in workers:
+        expected_actor_groups.add(frozenset([worker]))
+    assert actor_groups == expected_actor_groups
+
+    # Sanity check: No P2P communicator constructed.
```

Comment:
Change to "No P2P communicator created".

Reply:
ack